funcx_sdk/funcx/utils/url_parsing.py

```diff
@@ -0,0 +1,32 @@
+from urllib.parse import urlparse
+
+from funcx.utils.errors import InvalidServiceAddress
+
+
+def validate_service_address(service_address):
+    try:
+        url_data = urlparse(service_address)
+        # port must be accessed to raise port value issues
+        url_data.port
+        if url_data.scheme != "http" and url_data.scheme != "https":
+            raise InvalidServiceAddress("Protocol must be HTTP/HTTPS")
+
+        if url_data.netloc is None or url_data.hostname is None:
+            raise InvalidServiceAddress("Address is malformed")
+    except Exception as e:
+        raise InvalidServiceAddress(f"Address is malformed - {e}")
+
+
+def ws_uri_from_service_address(service_address):
```

Comment:
Would be good to add some type hints, and a docstring.

Reply:
Oops, you are right I meant to do that